File: group_anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = {}
        for s in strs:
            count = [0] * 26
            for c in s:
                count[ord(c) - ord('a')] += 1
            fingerprint = tuple(count)
            if fingerprint not in res:
                res[fingerprint] = []
            res[fingerprint].append(s)
        return list(res.values())

File: groupAnagrams.js
class Solution {
  /**
   * @param {string[]} strs
   * @return {string[][]}
   */
  groupAnagrams(strs) {
    let res = {};

    for (let s of strs) {
      const countMap = new Array(26).fill(0);
      for (let c of s) {
        countMap[c.charCodeAt(0) - "a".charCodeAt(0)]++;
      }
      const fingerPrint = countMap.join(",");
      if (!(fingerPrint in res)) {
        res[fingerPrint] = [];
      }
      res[fingerPrint].push(s);
    }

    return Object.values(res);
  }
}
